language/move-lang/src/expansion/ast.rs

```diff
@@ -69,19 +70,30 @@ pub struct ModuleDefinition {
     /// `dependency_order` is the topological order/rank in the dependency graph.
     /// `dependency_order` is initialized at `0` and set in the uses pass
     pub dependency_order: usize,
+    pub immediate_neighbors: BTreeSet<Neighbor>,
```

Comment:
Looks good here! But it is a tad bit weird that technically something can be a friend and a use in the set. I might have liked that a map would have made it a bit more intentional that it can only be one or the other, but its not a big deal